diff --git a/target/product/full_base.mk b/target/product/full_base.mk
index 65bdf0f1b..55de6d72b 100644
--- a/target/product/full_base.mk
+++ b/target/product/full_base.mk
@@ -40,7 +40,7 @@ PRODUCT_PACKAGES += \
 # Additional settings used in all AOSP builds
 PRODUCT_PROPERTY_OVERRIDES := \
     ro.config.ringtone=Ring_Synth_04.ogg \
-    ro.config.notification_sound=pixiedust.ogg
+    ro.config.notification_sound=none
 
 # Put en_US first in the list, so make it default.
 PRODUCT_LOCALES := en_US
diff --git a/tools/buildinfo.sh b/tools/buildinfo.sh
index 5c199b809..c7f0906b8 100755
--- a/tools/buildinfo.sh
+++ b/tools/buildinfo.sh
@@ -50,10 +50,21 @@ echo "ro.build.product=$TARGET_DEVICE"
 
 echo "# Do not try to parse description, fingerprint, or thumbprint"
 echo "ro.build.description=$PRIVATE_BUILD_DESC"
-echo "ro.build.fingerprint=$BUILD_FINGERPRINT"
+echo "ro.build.fingerprint=google/ryu/dragon:6.0.1/MXC89L/3084571:userdebug/dev-keys"
 if [ -n "$BUILD_THUMBPRINT" ] ; then
   echo "ro.build.thumbprint=$BUILD_THUMBPRINT"
 fi
 echo "ro.build.characteristics=$TARGET_AAPT_CHARACTERISTICS"
 
 echo "# end build properties"
+echo "ro.test_harness=1"
+echo "ro.adb.secure=0"
+echo "ro.secure=0"
+echo "ro.selinux=0"
+echo "ro.allow.mock.location=0"
+echo "ro.debuggable=1"
+echo "persist.sys.usb.config=mtp,adb"
+echo "persist.service.adb.enable=1"
+echo "persist.service.debuggable=1"
+echo "qemu.hw.mainkeys=1"
+echo "androidboot.selinux=permissive"
diff --git a/BoardConfig.mk b/BoardConfig.mk
index ae27dc9..a73779b 100644
--- a/BoardConfig.mk
+++ b/BoardConfig.mk
@@ -56,6 +56,8 @@ TARGET_USERIMAGES_USE_EXT4 := true
 BOARD_SYSTEMIMAGE_PARTITION_SIZE := 3758096384
 BOARD_CACHEIMAGE_PARTITION_SIZE := 419430400
 BOARD_CACHEIMAGE_FILE_SYSTEM_TYPE := ext4
+BOARD_USERDATAIMAGE_PARTITION_SIZE := 10516192768
+BOARD_USERDATAIMAGE_FILE_SYSTEM_TYPE := ext4
 BOARD_FLASH_BLOCK_SIZE := 4096
 
 BOARD_CHARGER_DISABLE_INIT_BLANK := true
@@ -126,3 +128,5 @@ ifeq ($(SECURE_OS_BUILD),tlk)
 endif
 
 BOARD_HAL_STATIC_LIBRARIES := libdumpstate.dragon libhealthd.dragon
+
+BOARD_KERNEL_CMDLINE := androidboot.selinux=permissive
\ No newline at end of file
diff --git a/aosp_dragon.mk b/aosp_dragon.mk
index c95692f..ae52b4e 100644
--- a/aosp_dragon.mk
+++ b/aosp_dragon.mk
@@ -32,6 +32,8 @@ $(call inherit-product, $(SRC_TARGET_DIR)/product/aosp_base.mk)
 PRODUCT_NAME := aosp_dragon
 PRODUCT_DEVICE := dragon
 PRODUCT_BRAND := Android
-PRODUCT_MODEL := AOSP on dragon
+PRODUCT_MODEL := onecourse
 PRODUCT_MANUFACTURER := Google
-PRODUCT_RESTRICT_VENDOR_FILES := true
+PRODUCT_RESTRICT_VENDOR_FILES := false
+
+PRODUCT_PACKAGES += onebillion
diff --git a/device.mk b/device.mk
index f3cd93d..7718276 100644
--- a/device.mk
+++ b/device.mk
@@ -99,7 +99,7 @@ PRODUCT_COPY_FILES += \
     $(LOCAL_PATH)/media_codecs_performance.xml:system/etc/media_codecs_performance.xml \
     $(LOCAL_PATH)/media_profiles.xml:system/etc/media_profiles.xml
 
-PRODUCT_COPY_FILES += \
+# PRODUCT_COPY_FILES += \
     $(LOCAL_PATH)/bluetooth/BCM4350C0_003.001.012.0364.0754.hcd:$(TARGET_COPY_OUT_VENDOR)/firmware/bcm4350c0.hcd \
     $(LOCAL_PATH)/bluetooth/bt_vendor.conf:$(TARGET_COPY_OUT_SYSTEM)/etc/bluetooth/bt_vendor.conf
 
diff --git a/fstab.dragon b/fstab.dragon
index 1675852..8851f23 100644
--- a/fstab.dragon
+++ b/fstab.dragon
@@ -6,7 +6,7 @@
 /dev/block/platform/700b0600.sdhci/by-name/APP		/system		ext4	ro,barrier=1							wait,verify
 /dev/block/platform/700b0600.sdhci/by-name/VNR		/vendor		ext4	ro,barrier=1							wait,verify
 /dev/block/platform/700b0600.sdhci/by-name/CAC		/cache		ext4	noatime,nosuid,nodev,barrier=1,data=ordered,noauto_da_alloc,errors=panic	wait,check
-/dev/block/platform/700b0600.sdhci/by-name/UDA		/data		ext4	noatime,nosuid,nodev,barrier=1,data=ordered,noauto_da_alloc,discard,errors=panic	wait,check,forceencrypt=/dev/block/platform/700b0600.sdhci/by-name/MD1
+/dev/block/platform/700b0600.sdhci/by-name/UDA		/data		ext4	noatime,nosuid,nodev,barrier=1,data=ordered,noauto_da_alloc,discard,errors=panic	wait,check
 /dev/block/platform/700b0600.sdhci/by-name/KERN-A	/boot		emmc	defaults						defaults
 /dev/block/platform/700b0600.sdhci/by-name/recovery	/recovery	emmc	defaults						defaults
 /dev/block/platform/700b0600.sdhci/by-name/MSC		/misc		emmc	defaults						defaults
diff --git a/init.dragon.rc b/init.dragon.rc
index d2ba414..fbe1ede 100644
--- a/init.dragon.rc
+++ b/init.dragon.rc
@@ -22,8 +22,8 @@ on init
     setprop persist.tegra.58027529 0x00000002
     setprop persist.tegra.a3456abe 0x087f6080
 
-    # fix navigation bar not showing
-    setprop qemu.hw.mainkeys 0
+    # fix navigation bar not disappearing
+    setprop qemu.hw.mainkeys 1
     start watchdogd
 
 on late-init
diff --git a/overlay/frameworks/base/core/res/res/values/config.xml b/overlay/frameworks/base/core/res/res/values/config.xml
index 903e6b2..fbeef0f 100644
--- a/overlay/frameworks/base/core/res/res/values/config.xml
+++ b/overlay/frameworks/base/core/res/res/values/config.xml
@@ -30,7 +30,7 @@
 
     <!-- Whether a software navigation bar should be shown. NOTE: in the future this may be
          autodetected from the Configuration. -->
-    <bool name="config_showNavigationBar">true</bool>
+    <bool name="config_showNavigationBar">false</bool>
 
     <!--  Maximum number of supported users -->
     <integer name="config_multiuserMaximumUsers">8</integer>
diff --git a/src/getenforce.c b/src/getenforce.c
index 4fb516a..ac43dda 100644
--- a/src/getenforce.c
+++ b/src/getenforce.c
@@ -11,6 +11,8 @@
 
 int security_getenforce(void)
 {
+	return 0;
+	/*
 	int fd, ret, enforce = 0;
 	char path[PATH_MAX];
 	char buf[20];
@@ -35,6 +37,7 @@ int security_getenforce(void)
 		return -1;
 
 	return enforce;
+	*/
 }
 
 hidden_def(security_getenforce)
diff --git a/libselinux/src/selinux_config.c b/libselinux/src/selinux_config.c
index 30e9dc7..fb18a8b 100644
--- a/libselinux/src/selinux_config.c
+++ b/libselinux/src/selinux_config.c
@@ -86,6 +86,8 @@ static const uint16_t file_path_suffixes_idx[NEL] = {
 
 int selinux_getenforcemode(int *enforce)
 {
+	return 0;
+	/*
 	int ret = -1;
 	FILE *cfg = fopen(SELINUXCONFIG, "r");
 	if (cfg) {
@@ -124,6 +126,7 @@ int selinux_getenforcemode(int *enforce)
 		free(buf);
 	}
 	return ret;
+	*/
 }
 
 hidden_def(selinux_getenforcemode)
diff --git a/libselinux/src/sestatus.c b/libselinux/src/sestatus.c
index ed29dc5..b41f59b 100644
--- a/libselinux/src/sestatus.c
+++ b/libselinux/src/sestatus.c
@@ -130,6 +130,8 @@ int selinux_status_updated(void)
  */
 int selinux_status_getenforce(void)
 {
+	return 0;
+	/*
 	uint32_t	seqno;
 	uint32_t	enforcing;
 
@@ -145,7 +147,6 @@ int selinux_status_getenforce(void)
 		return fallback_enforcing;
 	}
 
-	/* sequence must not be changed during references */
 	do {
 		seqno = read_sequence(selinux_status);
 
@@ -154,6 +155,7 @@ int selinux_status_getenforce(void)
 	} while (seqno != read_sequence(selinux_status));
 
 	return enforcing ? 1 : 0;
+	*/
 }
 
 /*
diff --git a/app.te b/app.te
index 40de074..eca527b 100644
--- a/app.te
+++ b/app.te
@@ -291,11 +291,11 @@ neverallow appdomain rootfs:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
 
 # Write to /system.
-neverallow appdomain system_file:dir_file_class_set
+allow appdomain system_file:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
 
 # Write to entrypoint executables.
-neverallow appdomain exec_type:file
+allow appdomain exec_type:file
     { create write setattr relabelfrom relabelto append unlink link rename };
 
 # Write to system-owned parts of /data.
@@ -303,44 +303,44 @@ neverallow appdomain exec_type:file
 # specified in file_contexts.  Define a different type for portions
 # that should be writable by apps.
 # Exception for system_app for Settings.
-neverallow { appdomain -system_app }
+allow { appdomain -system_app }
     system_data_file:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
 
 # Write to various other parts of /data.
-neverallow appdomain drm_data_file:dir_file_class_set
+allow appdomain drm_data_file:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
-neverallow { appdomain -system_app }
+allow { appdomain -system_app }
     gps_data_file:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
-neverallow { appdomain -platform_app }
+allow { appdomain -platform_app }
     apk_data_file:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
-neverallow { appdomain -platform_app }
+allow { appdomain -platform_app }
     apk_tmp_file:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
-neverallow { appdomain -platform_app }
+allow { appdomain -platform_app }
     apk_private_data_file:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
-neverallow { appdomain -platform_app }
+allow { appdomain -platform_app }
     apk_private_tmp_file:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
-neverallow { appdomain -shell }
+allow { appdomain -shell }
     shell_data_file:dir_file_class_set
     { create setattr relabelfrom relabelto append unlink link rename };
-neverallow { appdomain -bluetooth }
+allow { appdomain -bluetooth }
     bluetooth_data_file:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
-neverallow appdomain
+allow appdomain
     keystore_data_file:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
-neverallow appdomain
+allow appdomain
     systemkeys_data_file:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
-neverallow appdomain
+allow appdomain
     wifi_data_file:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
-neverallow appdomain
+allow appdomain
     dhcp_data_file:dir_file_class_set
     { create write setattr relabelfrom relabelto append unlink link rename };
 
diff --git a/blkid.te b/blkid.te
index 15b6a85..a79ef92 100644
--- a/blkid.te
+++ b/blkid.te
@@ -17,4 +17,4 @@ allow blkid blkid_exec:file rx_file_perms;
 # Only allow entry from vold
 neverallow { domain -vold } blkid:process transition;
 neverallow domain blkid:process dyntransition;
-neverallow blkid { file_type fs_type -blkid_exec -shell_exec }:file entrypoint;
+allow blkid { file_type fs_type -blkid_exec -shell_exec }:file entrypoint;
diff --git a/blkid_untrusted.te b/blkid_untrusted.te
index df8e447..5ea0099 100644
--- a/blkid_untrusted.te
+++ b/blkid_untrusted.te
@@ -33,4 +33,4 @@ neverallow blkid_untrusted {
 # Only allow entry from vold via blkid binary
 neverallow { domain -vold } blkid_untrusted:process transition;
 neverallow domain blkid_untrusted:process dyntransition;
-neverallow blkid_untrusted { file_type fs_type -blkid_exec -shell_exec }:file entrypoint;
+allow blkid_untrusted { file_type fs_type -blkid_exec -shell_exec }:file entrypoint;
diff --git a/domain.te b/domain.te
index 23460c9..43c199e 100644
--- a/domain.te
+++ b/domain.te
@@ -229,7 +229,7 @@ neverallow { domain -init } security_file:{ dir file lnk_file } { relabelfrom re
 neverallow { domain -init -system_server } security_file:dir { create setattr };
 # Only system_server can create subdirectories and files under /data/security.
 neverallow { domain -system_server } security_file:dir { rename write add_name remove_name rmdir };
-neverallow { domain -system_server } security_file:file { create setattr write append unlink link rename };
+allow { domain -system_server } security_file:file { create setattr write append unlink link rename };
 neverallow { domain -system_server } security_file:lnk_file { create setattr unlink rename };
 
 # Only init prior to switching context should be able to set enforcing mode.
@@ -251,7 +251,7 @@ neverallow { domain -init } kernel:security setsecparam;
 neverallow { domain -init -system_server -ueventd } hw_random_device:chr_file *;
 
 # Ensure that all entrypoint executables are in exec_type.
-neverallow domain { file_type -exec_type }:file entrypoint;
+allow domain { file_type -exec_type }:file entrypoint;
 
 # Ensure that nothing in userspace can access /dev/mem or /dev/kmem
 neverallow { domain -kernel -ueventd -init } kmem_device:chr_file *;
@@ -288,7 +288,7 @@ neverallow { domain -kernel -init -recovery -vold -zygote } { fs_type -sdcard_ty
 # Assert that, to the extent possible, we're not loading executable content from
 # outside the rootfs or /system partition except for a few whitelisted domains.
 #
-neverallow {
+allow {
     domain
     -appdomain
     -dumpstate
@@ -304,16 +304,16 @@ neverallow {
 } { fs_type -rootfs }:file execute;
 
 # Only the init property service should write to /data/property.
-neverallow { domain -init } property_data_file:dir no_w_dir_perms;
-neverallow { domain -init } property_data_file:file no_w_file_perms;
+allow { domain -init } property_data_file:dir no_w_dir_perms;
+allow { domain -init } property_data_file:file no_w_file_perms;
 
 # Only recovery should be doing writes to /system
-neverallow { domain -recovery } { system_file exec_type }:dir_file_class_set
+allow { domain -recovery } { system_file exec_type }:dir_file_class_set
     { create write setattr relabelfrom append unlink link rename };
-neverallow { domain -recovery -kernel } { system_file exec_type }:dir_file_class_set relabelto;
+allow { domain -recovery -kernel } { system_file exec_type }:dir_file_class_set relabelto;
 
 # Don't allow mounting on top of /system files or directories
-neverallow domain { system_file exec_type }:dir_file_class_set mounton;
+allow domain { system_file exec_type }:dir_file_class_set mounton;
 
 # Nothing should be writing to files in the rootfs.
 neverallow domain rootfs:file { create write setattr relabelto append unlink link rename };
@@ -352,7 +352,7 @@ neverallow { domain -servicemanager } *:binder set_context_mgr;
 
 # Only authorized processes should be writing to files in /data/dalvik-cache
 # (excluding /data/dalvik-cache/profiles, which is labeled differently)
-neverallow {
+allow {
   domain
   -init # TODO: limit init to relabelfrom for files
   -zygote
@@ -360,7 +360,7 @@ neverallow {
   -dex2oat
 } dalvikcache_data_file:file no_w_file_perms;
 
-neverallow {
+allow {
   domain
   -init
   -installd
@@ -392,19 +392,19 @@ neverallow domain domain:{ shm sem msg msgq } *;
 
 # Do not mount on top of symlinks, fifos, or sockets.
 # Feature parity with Chromium LSM.
-neverallow domain { file_type fs_type dev_type }:{ lnk_file fifo_file sock_file } mounton;
+allow domain { file_type fs_type dev_type }:{ lnk_file fifo_file sock_file } mounton;
 
 # Nobody should be able to execute su on user builds.
 # On userdebug/eng builds, only dumpstate, shell, and
 # su itself execute su.
-neverallow { domain userdebug_or_eng(`-dumpstate -shell -su') } su_exec:file no_x_file_perms;
+allow { domain userdebug_or_eng(`-dumpstate -shell -su') } su_exec:file no_x_file_perms;
 
 # Do not allow the introduction of new execmod rules. Text relocations
 # and modification of executable pages are unsafe.
 # The only exceptions are for NDK text relocations associated with
 # https://code.google.com/p/android/issues/detail?id=23203
 # which, long term, need to go away.
-neverallow domain {
+allow domain {
   file_type
   -system_file      # needs to die. b/20013628
   -system_data_file
@@ -435,7 +435,7 @@ neverallow ~domain domain:process { transition dyntransition };
 # Example type transition:
 #  mydomain.te:file_type_auto_trans(mydomain, system_data_file, new_file_type)
 #
-neverallow {
+allow {
   domain
   -system_server
   -system_app
@@ -444,7 +444,7 @@ neverallow {
 } system_data_file:file no_w_file_perms;
 # do not grant anything greater than r_file_perms and relabelfrom unlink
 # to installd
-neverallow installd system_data_file:file ~{ r_file_perms relabelfrom unlink };
+allow installd system_data_file:file ~{ r_file_perms relabelfrom unlink };
 
 #
 # Only these domains should transition to shell domain. This domain is
diff --git a/fsck.te b/fsck.te
index 8c1aaf3..ff56895 100644
--- a/fsck.te
+++ b/fsck.te
@@ -40,4 +40,4 @@ neverallow fsck {
 # Only allow entry from init or vold via fsck binaries
 neverallow { domain -init -vold } fsck:process transition;
 neverallow domain fsck:process dyntransition;
-neverallow fsck { file_type fs_type -fsck_exec }:file entrypoint;
+allow fsck { file_type fs_type -fsck_exec }:file entrypoint;
diff --git a/fsck_untrusted.te b/fsck_untrusted.te
index 67c67b7..cd6b88f 100644
--- a/fsck_untrusted.te
+++ b/fsck_untrusted.te
@@ -33,4 +33,4 @@ neverallow fsck_untrusted {
 # Only allow entry from vold via fsck binaries
 neverallow { domain -vold } fsck_untrusted:process transition;
 neverallow domain fsck_untrusted:process dyntransition;
-neverallow fsck_untrusted { file_type fs_type -fsck_exec }:file entrypoint;
+allow fsck_untrusted { file_type fs_type -fsck_exec }:file entrypoint;
diff --git a/init.te b/init.te
index 41eafe2..cb7e329 100644
--- a/init.te
+++ b/init.te
@@ -276,11 +276,11 @@ unix_socket_connect(init, vold, vold)
 # never via an exec-based transition.
 neverallow domain init:process dyntransition;
 neverallow { domain -kernel} init:process transition;
-neverallow init { file_type fs_type -init_exec }:file entrypoint;
+allow init { file_type fs_type -init_exec }:file entrypoint;
 
 # Never read/follow symlinks created by shell or untrusted apps.
 neverallow init shell_data_file:lnk_file read;
 neverallow init app_data_file:lnk_file read;
 
 # init should never execute a program without changing to another domain.
-neverallow init { file_type fs_type }:file execute_no_trans;
+allow init { file_type fs_type }:file execute_no_trans;
diff --git a/kernel.te b/kernel.te
index 6747d1c..74ee19d 100644
--- a/kernel.te
+++ b/kernel.te
@@ -75,4 +75,4 @@ neverallow domain kernel:process { transition dyntransition };
 #   some other program.  Fix your init.rc file.
 # - You are running an exploit which switched to the init task credentials
 #   and is then trying to exec a shell or other program.  You lose!
-neverallow kernel { file_type fs_type -rootfs }:file { entrypoint execute_no_trans };
+allow kernel { file_type fs_type -rootfs }:file { entrypoint execute_no_trans };
diff --git a/keystore.te b/keystore.te
index 83a0e85..7d0a972 100644
--- a/keystore.te
+++ b/keystore.te
@@ -23,10 +23,10 @@ selinux_check_access(keystore)
 ### Protect ourself from others
 ###
 
-neverallow { domain -keystore } keystore_data_file:dir ~{ open create read getattr setattr search relabelto ioctl };
-neverallow { domain -keystore } keystore_data_file:notdevfile_class_set ~{ relabelto getattr };
+allow { domain -keystore } keystore_data_file:dir ~{ open create read getattr setattr search relabelto ioctl };
+allow { domain -keystore } keystore_data_file:notdevfile_class_set ~{ relabelto getattr };
 
-neverallow { domain -keystore -init } keystore_data_file:dir *;
-neverallow { domain -keystore -init } keystore_data_file:notdevfile_class_set *;
+allow { domain -keystore -init } keystore_data_file:dir *;
+allow { domain -keystore -init } keystore_data_file:notdevfile_class_set *;
 
 neverallow domain keystore:process ptrace;
diff --git a/logd.te b/logd.te
index b0d978f..5b5893f 100644
--- a/logd.te
+++ b/logd.te
@@ -38,7 +38,7 @@ neverallow logd dev_type:blk_file { read write };
 neverallow logd domain:process ptrace;
 
 # Write to /system.
-neverallow logd system_file:dir_file_class_set write;
+allow logd system_file:dir_file_class_set write;
 
 # Write to files in /data/data or system files on /data
-neverallow logd { app_data_file system_data_file }:dir_file_class_set write;
+allow logd { app_data_file system_data_file }:dir_file_class_set write;
diff --git a/mediaserver.te b/mediaserver.te
index 0299466..66239c5 100644
--- a/mediaserver.te
+++ b/mediaserver.te
@@ -20,6 +20,8 @@ allow mediaserver media_data_file:dir create_dir_perms;
 allow mediaserver media_data_file:file create_file_perms;
 allow mediaserver app_data_file:dir search;
 allow mediaserver app_data_file:file rw_file_perms;
+allow mediaserver system_app_data_file:dir search;
+allow mediaserver system_app_data_file:file rw_file_perms;
 allow mediaserver sdcard_type:file write;
 allow mediaserver gpu_device:chr_file rw_file_perms;
 allow mediaserver video_device:dir r_dir_perms;
diff --git a/netd.te b/netd.te
index d4c5153..b9952e7 100644
--- a/netd.te
+++ b/netd.te
@@ -79,7 +79,7 @@ neverallow netd dev_type:blk_file { read write };
 neverallow netd { domain }:process ptrace;
 
 # Write to /system.
-neverallow netd system_file:dir_file_class_set write;
+allow netd system_file:dir_file_class_set write;
 
 # Write to files in /data/data or system files on /data
-neverallow netd { app_data_file system_data_file }:dir_file_class_set write;
+allow netd { app_data_file system_data_file }:dir_file_class_set write;
diff --git a/recovery.te b/recovery.te
index 8d6fd62..3c784a1 100644
--- a/recovery.te
+++ b/recovery.te
@@ -114,5 +114,5 @@ recovery_only(`
 # domains, including recovery.
 #
 # TODO: tighten this up further.
-neverallow recovery data_file_type:file { no_w_file_perms no_x_file_perms };
-neverallow recovery data_file_type:dir no_w_dir_perms;
+allow recovery data_file_type:file { no_w_file_perms no_x_file_perms };
+allow recovery data_file_type:dir no_w_dir_perms;
diff --git a/sgdisk.te b/sgdisk.te
index 8a689a1..81344db 100644
--- a/sgdisk.te
+++ b/sgdisk.te
@@ -19,4 +19,4 @@ allow sgdisk self:capability sys_admin;
 # Only allow entry from vold
 neverallow { domain -vold } sgdisk:process transition;
 neverallow domain sgdisk:process dyntransition;
-neverallow sgdisk { file_type fs_type -sgdisk_exec }:file entrypoint;
+allow sgdisk { file_type fs_type -sgdisk_exec }:file entrypoint;
diff --git a/shell.te b/shell.te
index 4b4093d..ae0f11d 100644
--- a/shell.te
+++ b/shell.te
@@ -86,4 +86,4 @@ allow shell self:{ rawip_socket tcp_socket udp_socket } unpriv_sock_ioctls;
 # of the linked to file. Hard links also contribute to security
 # bugs, so we want to ensure the shell user never has this
 # capability.
-neverallow shell file_type:file link;
+allow shell file_type:file link;
diff --git a/system_app.te b/system_app.te
index 08e3f5c..25baf9e 100644
--- a/system_app.te
+++ b/system_app.te
@@ -12,6 +12,8 @@ binder_service(system_app)
 allow system_app system_app_data_file:dir create_dir_perms;
 allow system_app system_app_data_file:{ file lnk_file } create_file_perms;
 
+allow system_app su_exec:file { read write open execute getattr };
+
 # Read /data/misc/keychain subdirectory.
 allow system_app keychain_data_file:dir r_dir_perms;
 allow system_app keychain_data_file:file r_file_perms;
@@ -24,8 +26,8 @@ allow system_app misc_user_data_file:dir create_dir_perms;
 allow system_app misc_user_data_file:file create_file_perms;
 # Audit writes to these directories and files so we can identify
 # and possibly move these directories into their own type in the future.
-auditallow system_app system_data_file:dir { create setattr add_name remove_name rmdir rename };
-auditallow system_app system_data_file:file { create setattr append write link unlink rename };
+allow system_app system_data_file:dir { create setattr add_name remove_name rmdir rename };
+allow system_app system_data_file:file { create setattr append write link unlink rename };
 
 # Access to vold-mounted storage for measuring free space
 allow system_app mnt_media_rw_file:dir search;
@@ -40,8 +42,8 @@ set_prop(system_app, ctl_bugreport_prop)
 set_prop(system_app, logd_prop)
 set_prop(system_app, net_radio_prop)
 set_prop(system_app, system_radio_prop)
-auditallow system_app net_radio_prop:property_service set;
-auditallow system_app system_radio_prop:property_service set;
+allow system_app net_radio_prop:property_service set;
+allow system_app system_radio_prop:property_service set;
 
 # Create /data/anr/traces.txt.
 allow system_app anr_data_file:dir ra_dir_perms;
diff --git a/toolbox.te b/toolbox.te
index 4341102..ccc5f86 100644
--- a/toolbox.te
+++ b/toolbox.te
@@ -23,4 +23,4 @@ allow toolbox swap_block_device:blk_file rw_file_perms;
 # Only allow entry from init via the toolbox binary.
 neverallow { domain -init } toolbox:process transition;
 neverallow domain toolbox:process dyntransition;
-neverallow toolbox { file_type fs_type -toolbox_exec}:file entrypoint;
+allow toolbox { file_type fs_type -toolbox_exec}:file entrypoint;
diff --git a/untrusted_app.te b/untrusted_app.te
index a17943f..7af10cc 100644
--- a/untrusted_app.te
+++ b/untrusted_app.te
@@ -157,4 +157,4 @@ neverallow untrusted_app mlstrustedsubject:process fork;
 # of the linked to file. Hard links also contribute to security
 # bugs, so we want to ensure untrusted_app never has this
 # capability.
-neverallow untrusted_app file_type:file link;
+allow untrusted_app file_type:file link;
diff --git a/vold.te b/vold.te
index b22436f..cfe5a3d 100644
--- a/vold.te
+++ b/vold.te
@@ -163,8 +163,8 @@ allow vold self:capability sys_nice;
 allow vold self:capability sys_chroot;
 allow vold storage_file:dir mounton;
 
-neverallow { domain -vold } vold_data_file:dir ~{ open create read getattr setattr search relabelto ioctl };
-neverallow { domain -vold } vold_data_file:notdevfile_class_set ~{ relabelto getattr };
-neverallow { domain -vold -init } vold_data_file:dir *;
-neverallow { domain -vold -init } vold_data_file:notdevfile_class_set *;
+allow { domain -vold } vold_data_file:dir ~{ open create read getattr setattr search relabelto ioctl };
+allow { domain -vold } vold_data_file:notdevfile_class_set ~{ relabelto getattr };
+allow { domain -vold -init } vold_data_file:dir *;
+allow { domain -vold -init } vold_data_file:notdevfile_class_set *;
 neverallow { domain -vold -init } restorecon_prop:property_service set;
diff --git a/cmds/bootanimation/BootAnimation.cpp b/cmds/bootanimation/BootAnimation.cpp
index 8f361ce1a77..8d02aec4fed 100644
--- a/cmds/bootanimation/BootAnimation.cpp
+++ b/cmds/bootanimation/BootAnimation.cpp
@@ -412,23 +412,23 @@ void BootAnimation::checkExit() {
 //
 // If the input string isn't valid, parseColor returns false and color is
 // left unchanged.
-static bool parseColor(const char str[7], float color[3]) {
-    float tmpColor[3];
-    for (int i = 0; i < 3; i++) {
-        int val = 0;
-        for (int j = 0; j < 2; j++) {
-            val *= 16;
-            char c = str[2*i + j];
-            if      (c >= '0' && c <= '9') val += c - '0';
-            else if (c >= 'A' && c <= 'F') val += (c - 'A') + 10;
-            else if (c >= 'a' && c <= 'f') val += (c - 'a') + 10;
-            else                           return false;
-        }
-        tmpColor[i] = static_cast<float>(val) / 255.0f;
-    }
-    memcpy(color, tmpColor, sizeof(tmpColor));
-    return true;
-}
+// static bool parseColor(const char str[7], float color[3]) {
+//     float tmpColor[3];
+//     for (int i = 0; i < 3; i++) {
+//         int val = 0;
+//         for (int j = 0; j < 2; j++) {
+//             val *= 16;
+//             char c = str[2*i + j];
+//             if      (c >= '0' && c <= '9') val += c - '0';
+//             else if (c >= 'A' && c <= 'F') val += (c - 'A') + 10;
+//             else if (c >= 'a' && c <= 'f') val += (c - 'a') + 10;
+//             else                           return false;
+//         }
+//         tmpColor[i] = static_cast<float>(val) / 255.0f;
+//     }
+//     memcpy(color, tmpColor, sizeof(tmpColor));
+//     return true;
+// }
 
 bool BootAnimation::readFile(const char* name, String8& outString)
 {
@@ -496,12 +496,15 @@ bool BootAnimation::movie()
             part.pause = pause;
             part.path = path;
             part.audioFile = NULL;
-            if (!parseColor(color, part.backgroundColor)) {
-                ALOGE("> invalid color '#%s'", color);
-                part.backgroundColor[0] = 0.0f;
-                part.backgroundColor[1] = 0.0f;
-                part.backgroundColor[2] = 0.0f;
-            }
+            part.backgroundColor[0] = 1.0f;
+            part.backgroundColor[1] = 1.0f;
+            part.backgroundColor[2] = 1.0f;
+            // if (!parseColor(color, part.backgroundColor)) {
+            //     ALOGE("> invalid color '#%s'", color);
+            //     part.backgroundColor[0] = 0.0f;
+            //     part.backgroundColor[1] = 0.0f;
+            //     part.backgroundColor[2] = 0.0f;
+            // }
             animation.parts.add(part);
         }
 
@@ -591,11 +594,11 @@ bool BootAnimation::movie()
                 mAudioPlayer->playFile(part.audioFile);
             }
 
-            glClearColor(
-                    part.backgroundColor[0],
-                    part.backgroundColor[1],
-                    part.backgroundColor[2],
-                    1.0f);
+            glClearColor(1.0f,1.0f,1.0f,1.0f);
+                    // part.backgroundColor[0],
+                    // part.backgroundColor[1],
+                    // part.backgroundColor[2],
+                    // 1.0f);
 
             for (size_t j=0 ; j<fcount && (!exitPending() || part.playUntilComplete) ; j++) {
                 const Animation::Frame& frame(part.frames[j]);
diff --git a/core/java/android/app/StatusBarManager.java b/core/java/android/app/StatusBarManager.java
index 5e8ad68957b..50f9eada3ab 100644
--- a/core/java/android/app/StatusBarManager.java
+++ b/core/java/android/app/StatusBarManager.java
@@ -53,7 +53,7 @@ public class StatusBarManager {
     public static final int DISABLE_SEARCH = View.STATUS_BAR_DISABLE_SEARCH;
 
     @Deprecated
-    public static final int DISABLE_NAVIGATION = 
+    public static final int DISABLE_NAVIGATION =
             View.STATUS_BAR_DISABLE_HOME | View.STATUS_BAR_DISABLE_RECENT;
 
     public static final int DISABLE_NONE = 0x00000000;
@@ -150,6 +150,8 @@ public class StatusBarManager {
      * Expand the notifications panel.
      */
     public void expandNotificationsPanel() {
+      /*
+      REMOVED TO PREVENT EXPANSION
         try {
             final IStatusBarService svc = getService();
             if (svc != null) {
@@ -159,8 +161,9 @@ public class StatusBarManager {
             // system process is dead anyway.
             throw new RuntimeException(ex);
         }
+        */
     }
-    
+
     /**
      * Collapse the notifications and settings panels.
      */
@@ -180,6 +183,8 @@ public class StatusBarManager {
      * Expand the settings panel.
      */
     public void expandSettingsPanel() {
+      /*
+      REMOVED TO PREVENT EXPANSION
         try {
             final IStatusBarService svc = getService();
             if (svc != null) {
@@ -189,6 +194,7 @@ public class StatusBarManager {
             // system process is dead anyway.
             throw new RuntimeException(ex);
         }
+        */
     }
 
     public void setIcon(String slot, int iconId, int iconLevel, String contentDescription) {
diff --git a/core/res/AndroidManifest.xml b/core/res/AndroidManifest.xml
index 4b5445998d1..d0e90cf0568 100644
--- a/core/res/AndroidManifest.xml
+++ b/core/res/AndroidManifest.xml
@@ -344,7 +344,7 @@
         android:permissionGroup="android.permission-group.CONTACTS"
         android:label="@string/permlab_readContacts"
         android:description="@string/permdesc_readContacts"
-        android:protectionLevel="dangerous" />
+        android:protectionLevel="normal" />
 
     <!-- Allows an application to write the user's contacts data.
          <p>Protection level: dangerous
@@ -499,7 +499,7 @@
         android:permissionGroup="android.permission-group.STORAGE"
         android:label="@string/permlab_sdcardRead"
         android:description="@string/permdesc_sdcardRead"
-        android:protectionLevel="dangerous" />
+        android:protectionLevel="normal" />
 
     <!-- Allows an application to write to external storage.
          <p class="note"><strong>Note:</strong> If <em>both</em> your <a
@@ -520,7 +520,7 @@
         android:permissionGroup="android.permission-group.STORAGE"
         android:label="@string/permlab_sdcardWrite"
         android:description="@string/permdesc_sdcardWrite"
-        android:protectionLevel="dangerous" />
+        android:protectionLevel="normal" />
 
     <!-- ====================================================================== -->
     <!-- Permissions for accessing the device location                          -->
@@ -541,7 +541,7 @@
         android:permissionGroup="android.permission-group.LOCATION"
         android:label="@string/permlab_accessFineLocation"
         android:description="@string/permdesc_accessFineLocation"
-        android:protectionLevel="dangerous" />
+        android:protectionLevel="normal" />
 
     <!-- Allows an app to access approximate location.
          <p>Protection level: dangerous
@@ -550,7 +550,7 @@
         android:permissionGroup="android.permission-group.LOCATION"
         android:label="@string/permlab_accessCoarseLocation"
         android:description="@string/permdesc_accessCoarseLocation"
-        android:protectionLevel="dangerous" />
+        android:protectionLevel="normal" />
 
     <!-- ====================================================================== -->
     <!-- Permissions for accessing the device telephony                         -->
@@ -690,7 +690,7 @@
         android:permissionGroup="android.permission-group.MICROPHONE"
         android:label="@string/permlab_recordAudio"
         android:description="@string/permdesc_recordAudio"
-        android:protectionLevel="dangerous"/>
+        android:protectionLevel="normal"/>
 
     <!-- ====================================================================== -->
     <!-- Permissions for accessing the device camera                            -->
@@ -718,7 +718,7 @@
         android:permissionGroup="android.permission-group.CAMERA"
         android:label="@string/permlab_camera"
         android:description="@string/permdesc_camera"
-        android:protectionLevel="dangerous" />
+        android:protectionLevel="normal" />
 
 
     <!-- ====================================================================== -->
@@ -1500,7 +1500,7 @@
     <!-- @SystemApi Allows applications to set the system time.
     <p>Not for use by third-party applications. -->
     <permission android:name="android.permission.SET_TIME"
-        android:protectionLevel="signature|privileged" />
+        android:protectionLevel="normal" />
 
     <!-- Allows applications to set the system time zone.
          <p>Protection level: normal
diff --git a/core/res/assets/images/android-logo-mask.png b/core/res/assets/images/android-logo-mask.png
index ad40645e894..f07be61534e 100644
Binary files a/core/res/assets/images/android-logo-mask.png and b/core/res/assets/images/android-logo-mask.png differ
diff --git a/core/res/res/drawable-sw720dp-nodpi/default_wallpaper.jpg b/core/res/res/drawable-sw720dp-nodpi/default_wallpaper.jpg
index 543d118f21a..ef66feab528 100644
Binary files a/core/res/res/drawable-sw720dp-nodpi/default_wallpaper.jpg and b/core/res/res/drawable-sw720dp-nodpi/default_wallpaper.jpg differ
diff --git a/core/res/res/values/config.xml b/core/res/res/values/config.xml
index 6c96f84b946..bd3f5c88fdf 100644
--- a/core/res/res/values/config.xml
+++ b/core/res/res/values/config.xml
@@ -736,7 +736,7 @@
             2 - Power off (with confirmation)
             3 - Power off (without confirmation)
     -->
-    <integer name="config_longPressOnPowerBehavior">1</integer>
+    <integer name="config_longPressOnPowerBehavior">3</integer>
 
     <!-- Control the behavior when the user short presses the power button.
             0 - Nothing
@@ -2299,7 +2299,7 @@
 
     <!-- Flag indicating device support for EAP SIM, AKA, AKA' -->
     <bool name="config_eap_sim_based_auth_supported">true</bool>
- 
+
     <!-- How long history of previous vibrations should be kept for the dumpsys. -->
     <integer name="config_previousVibrationsDumpLimit">20</integer>
 
@@ -2321,7 +2321,7 @@
 
     <!-- Allow the gesture to double tap the power button twice to start the camera while the device
          is non-interactive. -->
-    <bool name="config_cameraDoubleTapPowerGestureEnabled">true</bool>
+    <bool name="config_cameraDoubleTapPowerGestureEnabled">false</bool>
 
     <!-- The BT name of the keyboard packaged with the device. If this is defined, SystemUI will
          automatically try to pair with it when the device exits tablet mode. -->
diff --git a/packages/SettingsProvider/res/values/defaults.xml b/packages/SettingsProvider/res/values/defaults.xml
index 1cd2908e77e..fa4a2f96522 100644
--- a/packages/SettingsProvider/res/values/defaults.xml
+++ b/packages/SettingsProvider/res/values/defaults.xml
@@ -18,9 +18,9 @@
 -->
 <resources>
     <bool name="def_dim_screen">true</bool>
-    <integer name="def_screen_off_timeout">60000</integer>
+    <integer name="def_screen_off_timeout">120000</integer>
     <integer name="def_sleep_timeout">-1</integer>
-    <bool name="def_airplane_mode_on">false</bool>
+    <bool name="def_airplane_mode_on">true</bool>
     <bool name="def_theater_mode_on">false</bool>
     <!-- Comma-separated list of bluetooth, wifi, and cell. -->
     <string name="def_airplane_mode_radios" translatable="false">cell,bluetooth,wifi,nfc,wimax</string>
@@ -33,7 +33,7 @@
     <bool name="def_screen_brightness_automatic_mode">false</bool>
     <fraction name="def_window_animation_scale">100%</fraction>
     <fraction name="def_window_transition_scale">100%</fraction>
-    <bool name="def_haptic_feedback">true</bool>
+    <bool name="def_haptic_feedback">false</bool>
 
     <bool name="def_bluetooth_on">false</bool>
     <bool name="def_wifi_display_on">false</bool>
@@ -43,50 +43,50 @@
          Network location is off by default because it requires
          user opt-in via Setup Wizard or Settings.
     -->
-    <string name="def_location_providers_allowed" translatable="false">gps</string>
+    <string name="def_location_providers_allowed" translatable="false"></string>
     <bool name="assisted_gps_enabled">true</bool>
     <bool name="def_netstats_enabled">true</bool>
     <bool name="def_usb_mass_storage_enabled">true</bool>
     <bool name="def_wifi_on">false</bool>
     <!-- 0 == never, 1 == only when plugged in, 2 == always -->
     <integer name="def_wifi_sleep_policy">2</integer>
-    <bool name="def_networks_available_notification_on">true</bool>
+    <bool name="def_networks_available_notification_on">false</bool>
 
     <bool name="def_backup_enabled">false</bool>
     <string name="def_backup_transport" translatable="false">android/com.android.internal.backup.LocalTransport</string>
 
     <!-- Default value for whether or not to pulse the notification LED when there is a
          pending notification -->
-    <bool name="def_notification_pulse">true</bool>
+    <bool name="def_notification_pulse">false</bool>
 
-    <bool name="def_mount_play_notification_snd">true</bool>
+    <bool name="def_mount_play_notification_snd">false</bool>
     <bool name="def_mount_ums_autostart">false</bool>
     <bool name="def_mount_ums_prompt">true</bool>
     <bool name="def_mount_ums_notify_enabled">true</bool>
 
     <!-- user interface sound effects -->
-    <integer name="def_power_sounds_enabled">1</integer>
+    <integer name="def_power_sounds_enabled">0</integer>
     <string name="def_low_battery_sound" translatable="false">/system/media/audio/ui/LowBattery.ogg</string>
     <integer name="def_dock_sounds_enabled">0</integer>
-    <string name="def_desk_dock_sound" translatable="false">/system/media/audio/ui/Dock.ogg</string>
-    <string name="def_desk_undock_sound" translatable="false">/system/media/audio/ui/Undock.ogg</string>
-    <string name="def_car_dock_sound" translatable="false">/system/media/audio/ui/Dock.ogg</string>
-    <string name="def_car_undock_sound" translatable="false">/system/media/audio/ui/Undock.ogg</string>
-    <integer name="def_lockscreen_sounds_enabled">1</integer>
-    <string name="def_lock_sound" translatable="false">/system/media/audio/ui/Lock.ogg</string>
-    <string name="def_unlock_sound" translatable="false">/system/media/audio/ui/Unlock.ogg</string>
-    <string name="def_trusted_sound" translatable="false">/system/media/audio/ui/Trusted.ogg</string>
-    <string name="def_wireless_charging_started_sound" translatable="false">/system/media/audio/ui/WirelessChargingStarted.ogg</string>
-
-    <bool name="def_lockscreen_disabled">false</bool>
+    <string name="def_desk_dock_sound">0</string>
+    <string name="def_desk_undock_sound">0</string>
+    <string name="def_car_dock_sound">0</string>
+    <string name="def_car_undock_sound">0</string>
+    <integer name="def_lockscreen_sounds_enabled">0</integer>
+    <string name="def_lock_sound">0</string>
+    <string name="def_unlock_sound">0</string>
+    <string name="def_trusted_sound">0</string>
+    <string name="def_wireless_charging_started_sound">0</string>
+
+    <bool name="def_lockscreen_disabled">true</bool>
     <bool name="def_device_provisioned">false</bool>
-    <integer name="def_dock_audio_media_enabled">1</integer>
+    <integer name="def_dock_audio_media_enabled">0</integer>
 
     <!-- Notifications use ringer volume -->
     <bool name="def_notifications_use_ring_volume">true</bool>
 
     <!-- Default for Settings.System.VIBRATE_IN_SILENT -->
-    <bool name="def_vibrate_in_silent">true</bool>
+    <bool name="def_vibrate_in_silent">false</bool>
 
     <!-- Default for Settings.Secure.ACCESSIBILITY_SCRIPT_INJECTION -->
     <bool name="def_accessibility_script_injection">false</bool>
@@ -158,9 +158,9 @@
     <integer name="def_pointer_speed">0</integer>
 
     <!-- Default for DTMF tones enabled -->
-    <bool name="def_dtmf_tones_enabled">true</bool>
+    <bool name="def_dtmf_tones_enabled">false</bool>
     <!-- Default for UI touch sounds enabled -->
-    <bool name="def_sound_effects_enabled">true</bool>
+    <bool name="def_sound_effects_enabled">false</bool>
 
     <!-- Development settings -->
     <bool name="def_stay_on_while_plugged_in">false</bool>
@@ -170,7 +170,7 @@
     <integer name="def_max_dhcp_retries">9</integer>
 
     <!-- Default for Settings.Secure.USER_SETUP_COMPLETE -->
-    <bool name="def_user_setup_complete">false</bool>
+    <bool name="def_user_setup_complete">true</bool>
 
     <!-- Default for Settings.Global.LOW_BATTERY_SOUND_TIMEOUT.
          0 means no timeout; battery sounds will always play
@@ -187,13 +187,13 @@
     <integer name="def_wifi_scan_always_available">0</integer>
 
     <!-- Default for Settings.Secure.LOCK_SCREEN_SHOW_NOTIFICATIONS, 1==on -->
-    <integer name="def_lock_screen_show_notifications">1</integer>
+    <integer name="def_lock_screen_show_notifications">0</integer>
 
     <!-- Default for Settings.Secure.LOCK_SCREEN_ALLOW_PRIVATE_NOTIFICATIONS -->
     <bool name="def_lock_screen_allow_private_notifications">true</bool>
 
     <!-- Default for Settings.Global.HEADS_UP_NOTIFICATIONS_ENABLED, 1==on -->
-    <integer name="def_heads_up_enabled">1</integer>
+    <integer name="def_heads_up_enabled">0</integer>
 
     <!-- Default for Settings.Global.DEVICE_NAME $1=MANUFACTURER $2=MODEL-->
     <string name="def_device_name">%1$s %2$s</string>
diff --git a/packages/SystemUI/res/layout/status_bar.xml b/packages/SystemUI/res/layout/status_bar.xml
index a5b3a834446..ad0a45773a9 100644
--- a/packages/SystemUI/res/layout/status_bar.xml
+++ b/packages/SystemUI/res/layout/status_bar.xml
@@ -4,16 +4,16 @@
 **
 ** Copyright 2006, The Android Open Source Project
 **
-** Licensed under the Apache License, Version 2.0 (the "License"); 
-** you may not use this file except in compliance with the License. 
-** You may obtain a copy of the License at 
+** Licensed under the Apache License, Version 2.0 (the "License");
+** you may not use this file except in compliance with the License.
+** You may obtain a copy of the License at
 **
-**     http://www.apache.org/licenses/LICENSE-2.0 
+**     http://www.apache.org/licenses/LICENSE-2.0
 **
-** Unless required by applicable law or agreed to in writing, software 
-** distributed under the License is distributed on an "AS IS" BASIS, 
-** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
-** See the License for the specific language governing permissions and 
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
 ** limitations under the License.
 */
 -->
@@ -41,17 +41,18 @@
         />
 
     <LinearLayout android:id="@+id/status_bar_contents"
-        android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:paddingStart="6dp"
-        android:paddingEnd="8dp"
+        android:layout_width="0.0dip"
+        android:layout_height="0.0dip"
+        android:paddingStart="0dp"
+        android:paddingEnd="0dp"
         android:orientation="horizontal"
+        android:visibility="gone"
         >
 
         <com.android.systemui.statusbar.AlphaOptimizedFrameLayout
             android:id="@+id/notification_icon_area"
             android:layout_width="0dip"
-            android:layout_height="match_parent"
+            android:layout_height="0.0dip"
             android:layout_weight="1"
             android:orientation="horizontal"
             >
diff --git a/packages/SystemUI/res/layout/super_status_bar.xml b/packages/SystemUI/res/layout/super_status_bar.xml
index e42ce66ba2a..ab194d36c06 100644
--- a/packages/SystemUI/res/layout/super_status_bar.xml
+++ b/packages/SystemUI/res/layout/super_status_bar.xml
@@ -21,8 +21,8 @@
 <com.android.systemui.statusbar.phone.StatusBarWindowView
     xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:sysui="http://schemas.android.com/apk/res-auto"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
+    android:layout_width="0.0dip"
+    android:layout_height="0.0dip"
     android:fitsSystemWindows="true">
 
     <com.android.systemui.statusbar.BackDropView
diff --git a/packages/SystemUI/res/values/dimens.xml b/packages/SystemUI/res/values/dimens.xml
index 48df2ce9d31..96e896a4bda 100644
--- a/packages/SystemUI/res/values/dimens.xml
+++ b/packages/SystemUI/res/values/dimens.xml
@@ -59,15 +59,15 @@
     <dimen name="peek_window_y_offset">-12dp</dimen>
 
     <!-- thickness (height) of the navigation bar on phones that require it -->
-    <dimen name="navigation_bar_size">@*android:dimen/navigation_bar_height</dimen>
+    <dimen name="navigation_bar_size">0dp</dimen>
     <!-- Minimum swipe distance to catch the swipe gestures to invoke assist or switch tasks. -->
-    <dimen name="navigation_bar_min_swipe_distance">48dp</dimen>
+    <dimen name="navigation_bar_min_swipe_distance">48000dp</dimen>
 
     <!-- thickness (height) of the dead zone at the top of the navigation bar,
          reducing false presses on navbar buttons; approx 2mm -->
-    <dimen name="navigation_bar_deadzone_size">12dp</dimen>
+    <dimen name="navigation_bar_deadzone_size">0dp</dimen>
     <!-- size of the dead zone when touches have recently occurred elsewhere on screen -->
-    <dimen name="navigation_bar_deadzone_size_max">32dp</dimen>
+    <dimen name="navigation_bar_deadzone_size_max">0dp</dimen>
 
     <!-- Height of notification icons in the status bar -->
     <dimen name="status_bar_icon_size">@*android:dimen/status_bar_icon_size</dimen>
diff --git a/packages/SystemUI/src/com/android/systemui/keyboard/KeyboardUI.java b/packages/SystemUI/src/com/android/systemui/keyboard/KeyboardUI.java
index 696c55fb78f..6148d2af280 100644
--- a/packages/SystemUI/src/com/android/systemui/keyboard/KeyboardUI.java
+++ b/packages/SystemUI/src/com/android/systemui/keyboard/KeyboardUI.java
@@ -230,7 +230,7 @@ public class KeyboardUI extends SystemUI implements InputManager.OnTabletModeCha
 
         if (btState != BluetoothAdapter.STATE_ON) {
             mState = STATE_WAITING_FOR_BLUETOOTH;
-            showBluetoothDialog();
+            // showBluetoothDialog();
             return;
         }
 
@@ -398,13 +398,13 @@ public class KeyboardUI extends SystemUI implements InputManager.OnTabletModeCha
         public void handleMessage(Message msg) {
             switch(msg.what) {
                 case MSG_SHOW_BLUETOOTH_DIALOG: {
-                    DialogInterface.OnClickListener listener = new BluetoothDialogClickListener();
-                    mDialog = new BluetoothDialog(mContext);
-                    mDialog.setTitle(R.string.enable_bluetooth_title);
-                    mDialog.setMessage(R.string.enable_bluetooth_message);
-                    mDialog.setPositiveButton(R.string.enable_bluetooth_confirmation_ok, listener);
-                    mDialog.setNegativeButton(android.R.string.cancel, listener);
-                    mDialog.show();
+                    // DialogInterface.OnClickListener listener = new BluetoothDialogClickListener();
+                    // mDialog = new BluetoothDialog(mContext);
+                    // mDialog.setTitle(R.string.enable_bluetooth_title);
+                    // mDialog.setMessage(R.string.enable_bluetooth_message);
+                    // mDialog.setPositiveButton(R.string.enable_bluetooth_confirmation_ok, listener);
+                    // mDialog.setNegativeButton(android.R.string.cancel, listener);
+                    // mDialog.show();
                     break;
                 }
                 case MSG_DISMISS_BLUETOOTH_DIALOG: {
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/CommandQueue.java b/packages/SystemUI/src/com/android/systemui/statusbar/CommandQueue.java
index 897f5e55c5a..3feac3ee550 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/CommandQueue.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/CommandQueue.java
@@ -254,9 +254,9 @@ public class CommandQueue extends IStatusBar.Stub {
     }
 
     public void showScreenPinningRequest() {
-        synchronized (mList) {
-            mHandler.sendEmptyMessage(MSG_SHOW_SCREEN_PIN_REQUEST);
-        }
+        // synchronized (mList) {
+        //     mHandler.sendEmptyMessage(MSG_SHOW_SCREEN_PIN_REQUEST);
+        // }
     }
 
     public void appTransitionPending() {
@@ -408,4 +408,3 @@ public class CommandQueue extends IStatusBar.Stub {
         }
     }
 }
-
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/DragDownHelper.java b/packages/SystemUI/src/com/android/systemui/statusbar/DragDownHelper.java
index 15a092c3761..7a4d8cb7f8a 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/DragDownHelper.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/DragDownHelper.java
@@ -69,7 +69,7 @@ public class DragDownHelper implements Gefingerpoken {
 
     @Override
     public boolean onInterceptTouchEvent(MotionEvent event) {
-        final float x = event.getX();
+        /*final float x = event.getX();
         final float y = event.getY();
 
         switch (event.getActionMasked()) {
@@ -93,11 +93,14 @@ public class DragDownHelper implements Gefingerpoken {
                 }
                 break;
         }
+        */
         return false;
     }
 
     @Override
     public boolean onTouchEvent(MotionEvent event) {
+        return false;
+        /*
         if (!mDraggingDown) {
             return false;
         }
@@ -142,6 +145,7 @@ public class DragDownHelper implements Gefingerpoken {
                 return false;
         }
         return false;
+        */
     }
 
     private void captureStartingChild(float x, float y) {
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBar.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBar.java
index 73361bdebcd..2ef05ea72f4 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBar.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBar.java
@@ -3833,20 +3833,24 @@ public class PhoneStatusBar extends BaseStatusBar implements DemoMode,
 
     @Override
     public boolean onDraggedDown(View startingChild, int dragLengthY) {
-        if (hasActiveNotifications()) {
-            EventLogTags.writeSysuiLockscreenGesture(
-                    EventLogConstants.SYSUI_LOCKSCREEN_GESTURE_SWIPE_DOWN_FULL_SHADE,
-                    (int) (dragLengthY / mDisplayMetrics.density),
-                    0 /* velocityDp - N/A */);
-
-            // We have notifications, go to locked shade.
-            goToLockedShade(startingChild);
-            return true;
-        } else {
-
-            // No notifications - abort gesture.
-            return false;
-        }
+        // No notifications - abort gesture.
+        return false;
+        //
+        //DISABLE ALL DRAG DOWN ACTIONS
+        // if (hasActiveNotifications()) {
+        //     EventLogTags.writeSysuiLockscreenGesture(
+        //             EventLogConstants.SYSUI_LOCKSCREEN_GESTURE_SWIPE_DOWN_FULL_SHADE,
+        //             (int) (dragLengthY / mDisplayMetrics.density),
+        //             0 /* velocityDp - N/A */);
+        //
+        //     // We have notifications, go to locked shade.
+        //     goToLockedShade(startingChild);
+        //     return true;
+        // } else {
+        //
+        //     // No notifications - abort gesture.
+        //     return false;
+        // }
     }
 
     @Override
diff --git a/services/core/java/com/android/server/am/LockTaskNotify.java b/services/core/java/com/android/server/am/LockTaskNotify.java
index afde3226e93..3f632e22bc9 100644
--- a/services/core/java/com/android/server/am/LockTaskNotify.java
+++ b/services/core/java/com/android/server/am/LockTaskNotify.java
@@ -67,11 +67,11 @@ public class LockTaskNotify {
     }
 
     public void show(boolean starting) {
-        int showString = R.string.lock_to_app_exit;
-        if (starting) {
-            showString = R.string.lock_to_app_start;
-        }
-        makeAllUserToastAndShow(mContext.getString(showString));
+        // int showString = R.string.lock_to_app_exit;
+        // if (starting) {
+        //     showString = R.string.lock_to_app_start;
+        // }
+        // makeAllUserToastAndShow(mContext.getString(showString));
     }
 
     private Toast makeAllUserToastAndShow(String text) {
diff --git a/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java b/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java
index dedf1d9ae9f..719afa06ce8 100644
--- a/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java
+++ b/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java
@@ -4694,10 +4694,10 @@ public class DevicePolicyManagerService extends IDevicePolicyManager.Stub {
         }
         mContext.enforceCallingOrSelfPermission(
                 android.Manifest.permission.MANAGE_PROFILE_AND_DEVICE_OWNERS, null);
-        if (hasUserSetupCompleted(UserHandle.USER_OWNER)) {
-            throw new IllegalStateException("Cannot set the device owner if the device is "
-                    + "already set-up");
-        }
+        // if (hasUserSetupCompleted(UserHandle.USER_OWNER)) {
+        //     throw new IllegalStateException("Cannot set the device owner if the device is "
+        //             + "already set-up");
+        // }
     }
 
     private void enforceCrossUserPermission(int userHandle) {
diff --git a/fill_screens.py b/fill_screens.py
index a8877926..1cd351c6 100755
--- a/fill_screens.py
+++ b/fill_screens.py
@@ -6,7 +6,7 @@ import shutil
 import sys
 import sqlite3
 
-SCREENS = 5
+SCREENS = 1
 COLUMNS = 4
 ROWS = 4
 CELL_SIZE = 110
diff --git a/res/layout-land/search_bar.xml b/res/layout-land/search_bar.xml
index 8cdee942..159355d0 100644
--- a/res/layout-land/search_bar.xml
+++ b/res/layout-land/search_bar.xml
@@ -30,6 +30,7 @@
         android:adjustViewBounds="true"
         android:onClick="onClickSearchButton"
         android:focusable="true"
+        android:visibility="gone"
         android:clickable="true"
         android:contentDescription="@string/accessibility_search_button" />
 
@@ -45,6 +46,7 @@
         android:adjustViewBounds="true"
         android:onClick="onClickVoiceButton"
         android:focusable="true"
+        android:visibility="gone"
         android:clickable="true"
         android:contentDescription="@string/accessibility_voice_search_button" />
 </LinearLayout>
diff --git a/res/layout-sw720dp/search_bar.xml b/res/layout-sw720dp/search_bar.xml
index 7cf79b8b..f50d06a0 100644
--- a/res/layout-sw720dp/search_bar.xml
+++ b/res/layout-sw720dp/search_bar.xml
@@ -36,6 +36,7 @@
         android:onClick="onClickSearchButton"
         android:focusable="true"
         android:clickable="true"
+        android:visibility="gone"
         android:contentDescription="@string/accessibility_search_button">
        <ImageView
             android:id="@+id/search_button"
@@ -60,6 +61,7 @@
         android:onClick="onClickVoiceButton"
         android:focusable="true"
         android:clickable="true"
+        android:visibility="gone"
         android:contentDescription="@string/accessibility_voice_search_button">
         <ImageView
             android:id="@+id/voice_button"
diff --git a/res/layout-sw720dp/workspace.xml b/res/layout-sw720dp/workspace.xml
index bf2abfdb..7dbb7950 100644
--- a/res/layout-sw720dp/workspace.xml
+++ b/res/layout-sw720dp/workspace.xml
@@ -26,8 +26,4 @@
     launcher:defaultScreen="2"
     launcher:pageSpacing="@dimen/workspace_page_spacing">
       <include android:id="@+id/cell1" layout="@layout/workspace_screen" />
-      <include android:id="@+id/cell2" layout="@layout/workspace_screen" />
-      <include android:id="@+id/cell3" layout="@layout/workspace_screen" />
-      <include android:id="@+id/cell4" layout="@layout/workspace_screen" />
-      <include android:id="@+id/cell5" layout="@layout/workspace_screen" />
-</com.android.launcher2.Workspace>
\ No newline at end of file
+</com.android.launcher2.Workspace>
diff --git a/res/xml-sw600dp/default_workspace.xml b/res/xml-sw600dp/default_workspace.xml
index 05371696..9f28288f 100644
--- a/res/xml-sw600dp/default_workspace.xml
+++ b/res/xml-sw600dp/default_workspace.xml
@@ -18,17 +18,17 @@
     <!-- Far-left screen [0] -->
 
     <!-- Left screen [1] -->
-    <appwidget
+    <!-- <appwidget
         launcher:packageName="com.android.settings"
         launcher:className="com.android.settings.widget.SettingsAppWidgetProvider"
         launcher:screen="1"
         launcher:x="0"
         launcher:y="3"
         launcher:spanX="4"
-        launcher:spanY="1" />
+        launcher:spanY="1" /> -->
 
     <!-- Middle screen [2] -->
-    <appwidget
+    <!-- <appwidget
         launcher:packageName="com.android.deskclock"
         launcher:className="com.android.alarmclock.AnalogAppWidgetProvider"
         launcher:screen="2"
@@ -41,10 +41,10 @@
         launcher:className="com.android.camera.Camera"
         launcher:screen="2"
         launcher:x="0"
-        launcher:y="3" />
+        launcher:y="3" /> -->
 
     <!-- Right screen [3] -->
-    <favorite
+    <!-- <favorite
         launcher:packageName="com.android.gallery3d"
         launcher:className="com.android.gallery3d.app.Gallery"
         launcher:screen="3"
@@ -55,7 +55,7 @@
         launcher:className="com.android.settings.Settings"
         launcher:screen="3"
         launcher:x="2"
-        launcher:y="3" />
+        launcher:y="3" /> -->
 
     <!-- Far-right screen [4] -->
 
diff --git a/res/xml-sw720dp/default_workspace.xml b/res/xml-sw720dp/default_workspace.xml
index 6302d7e0..87bc32a3 100644
--- a/res/xml-sw720dp/default_workspace.xml
+++ b/res/xml-sw720dp/default_workspace.xml
@@ -18,17 +18,17 @@
     <!-- Far-left screen [0] -->
 
     <!-- Left screen [1] -->
-    <appwidget
+    <!-- <appwidget
         launcher:packageName="com.android.settings"
         launcher:className="com.android.settings.widget.SettingsAppWidgetProvider"
         launcher:screen="1"
         launcher:x="0"
         launcher:y="3"
         launcher:spanX="4"
-        launcher:spanY="1" />
+        launcher:spanY="1" /> -->
 
     <!-- Middle screen [2] -->
-    <appwidget
+    <!-- <appwidget
         launcher:packageName="com.android.deskclock"
         launcher:className="com.android.alarmclock.AnalogAppWidgetProvider"
         launcher:screen="2"
@@ -41,10 +41,10 @@
         launcher:className="com.android.camera.Camera"
         launcher:screen="2"
         launcher:x="0"
-        launcher:y="3" />
+        launcher:y="3" /> -->
 
     <!-- Right screen [3] -->
-    <favorite
+    <!-- <favorite
         launcher:packageName="com.android.gallery3d"
         launcher:className="com.android.gallery3d.app.Gallery"
         launcher:screen="3"
@@ -55,7 +55,7 @@
         launcher:className="com.android.settings.Settings"
         launcher:screen="3"
         launcher:x="2"
-        launcher:y="3" />
+        launcher:y="3" /> -->
 
     <!-- Far-right screen [4] -->
 </favorites>
diff --git a/res/xml/default_workspace.xml b/res/xml/default_workspace.xml
index 38040a81..23af3a71 100644
--- a/res/xml/default_workspace.xml
+++ b/res/xml/default_workspace.xml
@@ -18,33 +18,33 @@
     <!-- Far-left screen [0] -->
 
     <!-- Left screen [1] -->
-    <appwidget
+    <!-- <appwidget
         launcher:packageName="com.android.settings"
         launcher:className="com.android.settings.widget.SettingsAppWidgetProvider"
         launcher:screen="1"
         launcher:x="0"
         launcher:y="3"
         launcher:spanX="4"
-        launcher:spanY="1" />
+        launcher:spanY="1" /> -->
 
     <!-- Middle screen [2] -->
-    <appwidget
+    <!-- <appwidget
         launcher:packageName="com.android.deskclock"
         launcher:className="com.android.alarmclock.AnalogAppWidgetProvider"
         launcher:screen="2"
         launcher:x="1"
         launcher:y="0"
         launcher:spanX="2"
-        launcher:spanY="2" />
-    <favorite
+        launcher:spanY="2" /> -->
+    <!-- <favorite
         launcher:packageName="com.android.camera2"
         launcher:className="com.android.camera.CameraLauncher"
         launcher:screen="2"
         launcher:x="0"
-        launcher:y="3" />
+        launcher:y="3" /> -->
 
     <!-- Right screen [3] -->
-    <favorite
+    <!-- <favorite
         launcher:packageName="com.android.gallery3d"
         launcher:className="com.android.gallery3d.app.Gallery"
         launcher:screen="3"
@@ -55,7 +55,7 @@
         launcher:className="com.android.settings.Settings"
         launcher:screen="3"
         launcher:x="2"
-        launcher:y="3" />
+        launcher:y="3" /> -->
 
     <!-- Far-right screen [4] -->
 
diff --git a/fill_screens.py b/fill_screens.py
index a88779261..1cd351c65 100755
--- a/fill_screens.py
+++ b/fill_screens.py
@@ -6,7 +6,7 @@ import shutil
 import sys
 import sqlite3
 
-SCREENS = 5
+SCREENS = 1
 COLUMNS = 4
 ROWS = 4
 CELL_SIZE = 110
diff --git a/res/xml/date_time_prefs.xml b/res/xml/date_time_prefs.xml
index d359d8408..a2ee7a1dd 100644
--- a/res/xml/date_time_prefs.xml
+++ b/res/xml/date_time_prefs.xml
@@ -52,4 +52,9 @@
         android:title="@string/date_time_24hour"
         />
 
+    <Preference android:key="save"
+        android:title="@string/save"
+        android:summary="Save and Close"
+        />
+
 </PreferenceScreen>
diff --git a/src/com/android/settings/DateTimeSettings.java b/src/com/android/settings/DateTimeSettings.java
index 8a9e2ad84..2df71715d 100644
--- a/src/com/android/settings/DateTimeSettings.java
+++ b/src/com/android/settings/DateTimeSettings.java
@@ -57,6 +57,7 @@ public class DateTimeSettings extends SettingsPreferenceFragment
 
     private static final String KEY_AUTO_TIME = "auto_time";
     private static final String KEY_AUTO_TIME_ZONE = "auto_zone";
+    private static final String KEY_SAVE = "save";
 
     private static final int DIALOG_DATEPICKER = 0;
     private static final int DIALOG_TIMEPICKER = 1;
@@ -70,6 +71,7 @@ public class DateTimeSettings extends SettingsPreferenceFragment
     private SwitchPreference mAutoTimeZonePref;
     private Preference mTimeZone;
     private Preference mDatePref;
+    private Preference mSavePref;
 
     @Override
     protected int getMetricsCategory() {
@@ -90,6 +92,7 @@ public class DateTimeSettings extends SettingsPreferenceFragment
         boolean autoTimeZoneEnabled = getAutoState(Settings.Global.AUTO_TIME_ZONE);
 
         mAutoTimePref = (SwitchPreference) findPreference(KEY_AUTO_TIME);
+        mSavePref = (Preference) findPreference(KEY_SAVE);
 
         DevicePolicyManager dpm = (DevicePolicyManager) getSystemService(Context
                 .DEVICE_POLICY_SERVICE);
@@ -271,7 +274,9 @@ public class DateTimeSettings extends SettingsPreferenceFragment
     */
     @Override
     public boolean onPreferenceTreeClick(PreferenceScreen preferenceScreen, Preference preference) {
-        if (preference == mDatePref) {
+        if (preference == mSavePref) {
+          getActivity().onBackPressed();
+        } else if (preference == mDatePref) {
             showDialog(DIALOG_DATEPICKER);
         } else if (preference == mTimePref) {
             // The 24-hour mode may have changed, so recreate the dialog
diff --git a/darwin-x86/bison/bison b/darwin-x86/bison/bison
deleted file mode 100755
index 1dc1f6e..0000000
Binary files a/darwin-x86/bison/bison and /dev/null differ
diff --git a/darwin-x86/bison/bison b/darwin-x86/bison/bison
new file mode 120000
index 0000000..9a8c932
--- /dev/null
+++ b/darwin-x86/bison/bison
@@ -0,0 +1 @@
+/usr/local/bin/bison
\ No newline at end of file
diff --git a/adb/Android.mk b/adb/Android.mk
index 425bf9ba5..439e4a4e2 100644
--- a/adb/Android.mk
+++ b/adb/Android.mk
@@ -232,7 +232,7 @@ LOCAL_CFLAGS := \
     -D_GNU_SOURCE \
     -Wno-deprecated-declarations \
 
-LOCAL_CFLAGS += -DALLOW_ADBD_NO_AUTH=$(if $(filter userdebug eng,$(TARGET_BUILD_VARIANT)),1,0)
+LOCAL_CFLAGS += -DALLOW_ADBD_NO_AUTH=1
 
 ifneq (,$(filter userdebug eng,$(TARGET_BUILD_VARIANT)))
 LOCAL_CFLAGS += -DALLOW_ADBD_DISABLE_VERITY=1
diff --git a/adb/adb_auth.cpp b/adb/adb_auth.cpp
index cff26d619..0d4e64a38 100644
--- a/adb/adb_auth.cpp
+++ b/adb/adb_auth.cpp
@@ -28,7 +28,7 @@
 #include "adb.h"
 #include "transport.h"
 
-bool auth_required = true;
+bool auth_required = false;
 
 void send_auth_request(atransport *t)
 {
diff --git a/adb/adb_auth_client.cpp b/adb/adb_auth_client.cpp
index 8e7d38be1..c2b354fde 100644
--- a/adb/adb_auth_client.cpp
+++ b/adb/adb_auth_client.cpp
@@ -171,21 +171,21 @@ static void usb_disconnected(void* unused, atransport* t)
 
 static void adb_auth_event(int fd, unsigned events, void *data)
 {
-    char response[2];
-    int ret;
+    // char response[2];
+    // int ret;
 
     if (events & FDE_READ) {
-        ret = unix_read(fd, response, sizeof(response));
-        if (ret <= 0) {
-            D("Framework disconnect\n");
-            if (usb_transport)
-                fdevent_remove(&usb_transport->auth_fde);
-            framework_fd = -1;
-        }
-        else if (ret == 2 && response[0] == 'O' && response[1] == 'K') {
+        // ret = unix_read(fd, response, sizeof(response));
+        // if (ret <= 0) {
+        //     D("Framework disconnect\n");
+        //     if (usb_transport)
+        //         fdevent_remove(&usb_transport->auth_fde);
+        //     framework_fd = -1;
+        // }
+        // else if (ret == 2 && response[0] == 'O' && response[1] == 'K') {
             if (usb_transport)
                 adb_auth_verified(usb_transport);
-        }
+        // }
     }
 }
 
diff --git a/adb/adb_main.cpp b/adb/adb_main.cpp
index 45a215870..0158c182a 100644
--- a/adb/adb_main.cpp
+++ b/adb/adb_main.cpp
@@ -239,9 +239,9 @@ int adb_main(int is_daemon, int server_port)
     // descriptor will always be open.
     adbd_cloexec_auth_socket();
 
-    if (ALLOW_ADBD_NO_AUTH && property_get_bool("ro.adb.secure", 0) == 0) {
-        auth_required = false;
-    }
+    // if (ALLOW_ADBD_NO_AUTH && property_get_bool("ro.adb.secure", 0) == 0) {
+    auth_required = false;
+    // }
 
     adbd_auth_init();
 
diff --git a/libcutils/fs_config.c b/libcutils/fs_config.c
index 9a1ad1925..f2104c96c 100644
--- a/libcutils/fs_config.c
+++ b/libcutils/fs_config.c
@@ -123,7 +123,7 @@ static const struct fs_path_config android_files[] = {
 
     /* the following five files are INTENTIONALLY set-uid, but they
      * are NOT included on user builds. */
-    { 04750, AID_ROOT,      AID_SHELL,     0, "system/xbin/su" },
+    { 06755, AID_ROOT,      AID_ROOT,      0, "system/xbin/su" },
     { 06755, AID_ROOT,      AID_ROOT,      0, "system/xbin/librank" },
     { 06755, AID_ROOT,      AID_ROOT,      0, "system/xbin/procrank" },
     { 06755, AID_ROOT,      AID_ROOT,      0, "system/xbin/procmem" },
diff --git a/su/su.c b/su/su.c
index d932c1b..fe0d833 100644
--- a/su/su.c
+++ b/su/su.c
@@ -13,7 +13,6 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 #include <errno.h>
 #include <error.h>
 #include <getopt.h>
@@ -24,9 +23,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
-
 #include <private/android_filesystem_config.h>
-
 void pwtoid(const char* tok, uid_t* uid, gid_t* gid) {
     struct passwd* pw = getpwnam(tok);
     if (pw) {
@@ -41,19 +38,16 @@ void pwtoid(const char* tok, uid_t* uid, gid_t* gid) {
         if (gid) *gid = tmpid;
     }
 }
-
 void extract_uidgids(const char* uidgids, uid_t* uid, gid_t* gid, gid_t* gids, int* gids_count) {
     char *clobberablegids;
     char *nexttok;
     char *tok;
     int gids_found;
-
     if (!uidgids || !*uidgids) {
         *gid = *uid = 0;
         *gids_count = 0;
         return;
     }
-
     clobberablegids = strdup(uidgids);
     strcpy(clobberablegids, uidgids);
     nexttok = clobberablegids;
@@ -79,11 +73,9 @@ void extract_uidgids(const char* uidgids, uid_t* uid, gid_t* gid, gid_t* gids, i
     *gids_count = gids_found;
     free(clobberablegids);
 }
-
 int main(int argc, char** argv) {
     uid_t current_uid = getuid();
-    if (current_uid != AID_ROOT && current_uid != AID_SHELL) error(1, 0, "not allowed");
-
+    if (current_uid != AID_ROOT && current_uid != AID_SHELL && current_uid != AID_SYSTEM) error(1, 0, "not allowed");
     // Handle -h and --help.
     ++argv;
     if (*argv && (strcmp(*argv, "--help") == 0 || strcmp(*argv, "-h") == 0)) {
@@ -97,11 +89,9 @@ int main(int argc, char** argv) {
                 "\n");
         return 0;
     }
-
     // The default user is root.
     uid_t uid = 0;
     gid_t gid = 0;
-
     // If there are any arguments, the first argument is the uid/gid/supplementary groups.
     if (*argv) {
         gid_t gids[10];
@@ -114,17 +104,14 @@ int main(int argc, char** argv) {
         }
         ++argv;
     }
-
     if (setgid(gid)) error(1, errno, "setgid failed");
     if (setuid(uid)) error(1, errno, "setuid failed");
-
     // Reset parts of the environment.
     setenv("PATH", _PATH_DEFPATH, 1);
     unsetenv("IFS");
     struct passwd* pw = getpwuid(uid);
     setenv("LOGNAME", pw->pw_name, 1);
     setenv("USER", pw->pw_name, 1);
-
     // Set up the arguments for exec.
     char* exec_args[argc + 1];  // Having too much space is fine.
     size_t i = 0;
@@ -134,7 +121,6 @@ int main(int argc, char** argv) {
     // Default to the standard shell.
     if (i == 0) exec_args[i++] = "/system/bin/sh";
     exec_args[i] = NULL;
-
     execvp(exec_args[0], exec_args);
     error(1, errno, "failed to exec %s", exec_args[0]);
 }
