Index: app/src/main/java/org/onebillion/xprz/mainui/x_lettersandsounds/X_Test.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/org/onebillion/xprz/mainui/x_lettersandsounds/X_Test.java	(revision )
+++ app/src/main/java/org/onebillion/xprz/mainui/x_lettersandsounds/X_Test.java	(revision )
@@ -0,0 +1,28 @@
+package org.onebillion.xprz.mainui.x_lettersandsounds;
+
+import android.graphics.PointF;
+import android.graphics.RectF;
+
+import org.onebillion.xprz.controls.OBControl;
+import org.onebillion.xprz.controls.OBGroup;
+import org.onebillion.xprz.controls.OBPath;
+import org.onebillion.xprz.mainui.XPRZ_SectionController;
+import org.onebillion.xprz.utils.OBUtils;
+import org.onebillion.xprz.utils.OB_Maths;
+
+/**
+ * Created by michal on 21/07/16.
+ */
+public class X_Test extends XPRZ_SectionController
+{
+
+    public void prepare()
+    {
+        super.prepare();
+        loadEvent("master");
+
+
+
+
+    }
+}
